### ISSUE TITLE
NRE in GetExplicitInterfaceImplementations when building dotnet/runtime

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#46575]
**Version Used**: 81c5d71

**Steps to Reproduce**:

`.\Build.cmd -pack -configuration Release` in Roslyn, then set up a dotnet/runtime enlistment to use your locally build compiler. You may use branch https://github.com/RikkiGibson/runtime/tree/use-local-roslyn as a starting point. NOTE: you must build a release package to repro this bug and change the `RestoreAdditionalProjectSources` path from Debug to Release.

Once that is complete, run `.\Build.cmd -subset libs` to repro the bug.

**Expected Behavior**:
Builds without errors

**Actual Behavior**:
```
Process terminated. System.NullReferenceException: Object reference not set to an instance of an object. [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PEMethodSymbol.get_ExplicitInterfaceImplementations() in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Metadata\PE\PEMethodSymbol.cs:line 1232 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SymbolExtensions.GetExplicitInterfaceImplementations(Symbol member) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\MemberSymbolExtensions.cs:line 516 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.MakeExplicitInterfaceImplementationMap() in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\TypeSymbol.cs:line 2091 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.GetExplicitImplementationForInterfaceMember(Symbol interfaceMember) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\TypeSymbol.cs:line 2082 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.ComputeImplementationForInterfaceMember(Symbol interfaceMember, TypeSymbol implementingType, DiagnosticBag diagnostics, Boolean ignoreImplementationInInterfaces, Boolean& implementationInInterfacesMightChangeResult) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\TypeSymbol.cs:line 839 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.ComputeImplementationAndDiagnosticsForInterfaceMember(Symbol interfaceMember, Boolean ignoreImplementationInInterfaces, Boolean& implementationInInterfacesMightChangeResult) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\TypeSymbol.cs:line 781 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.FindImplementationForInterfaceMemberInNonInterfaceWithDiagnostics(Symbol interfaceMember, Boolean ignoreImplementationInInterfacesIfResultIsNotReady) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\TypeSymbol.cs:line 757 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.ComputeInterfaceImplementations(DiagnosticBag diagnostics, CancellationToken cancellationToken) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Source\SourceMemberContainerSymbol_ImplementationChecks.cs:line 157 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetSynthesizedExplicitImplementations(CancellationToken cancellationToken) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Source\SourceMemberContainerSymbol_ImplementationChecks.cs:line 50 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.ForceComplete(SourceLocation locationOpt, CancellationToken cancellationToken) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Source\SourceMemberContainerSymbol.cs:line 507 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbol.ForceCompleteMemberByLocation(SourceLocation locationOpt, Symbol member, CancellationToken cancellationToken) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Symbol.cs:line 800 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamespaceSymbol.<>c__DisplayClass49_1.<ForceComplete>b__0(Int32 i) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Source\SourceNamespaceSymbol_Completion.cs:line 61 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Environment.FailFast(System.String, System.Exception) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.FailFast.OnFatalException(System.Exception) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.FatalError.Report(System.Exception, System.Action`1<System.Exception>) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.FatalError.Report(System.Exception) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.FatalError.ReportUnlessCanceled(System.Exception) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamespaceSymbol+<>c__DisplayClass49_1.<ForceComplete>b__0(Int32) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PEMethodSymbol.get_ExplicitInterfaceImplementations() [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SymbolExtensions.GetExplicitInterfaceImplementations(Microsoft.CodeAnalysis.CSharp.Symbol) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.MakeExplicitInterfaceImplementationMap() [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.GetExplicitImplementationForInterfaceMember(Microsoft.CodeAnalysis.CSharp.Symbol) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.ComputeImplementationForInterfaceMember(Microsoft.CodeAnalysis.CSharp.Symbol, Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol, Microsoft.CodeAnalysis.DiagnosticBag, Boolean, Boolean ByRef) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.ComputeImplementationAndDiagnosticsForInterfaceMember(Microsoft.CodeAnalysis.CSharp.Symbol, Boolean, Boolean ByRef) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.FindImplementationForInterfaceMemberInNonInterfaceWithDiagnostics(Microsoft.CodeAnalysis.CSharp.Symbol, Boolean) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.ComputeInterfaceImplementations(Microsoft.CodeAnalysis.DiagnosticBag, System.Threading.CancellationToken) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetSynthesizedExplicitImplementations(System.Threading.CancellationToken) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.ForceComplete(Microsoft.CodeAnalysis.SourceLocation, System.Threading.CancellationToken) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbol.ForceCompleteMemberByLocation(Microsoft.CodeAnalysis.SourceLocation, Microsoft.CodeAnalysis.CSharp.Symbol, System.Threading.CancellationToken) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamespaceSymbol+<>c__DisplayClass49_1.<ForceComplete>b__0(Int32) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Roslyn.Utilities.UICultureUtilities+<>c__DisplayClass6_0`1[[System.Int32, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<WithCurrentUICulture>b__0(Int32) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.Tasks.Parallel+<>c__DisplayClass19_0`1[[System.__Canon, System.Private.CoreLib, Version=5.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e]].<ForWorker>b__1(System.Threading.Tasks.RangeWorker ByRef, Int32, Boolean ByRef) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.Tasks.TaskReplicator+Replica`1[[System.Threading.Tasks.RangeWorker, System.Threading.Tasks.Parallel, Version=5.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a]].ExecuteAction(Boolean ByRef) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.Tasks.TaskReplicator+Replica.Execute() [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.Tasks.TaskReplicator+Replica+<>c.<.ctor>b__4_0(System.Object) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.Tasks.Task.InnerInvoke() [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.Tasks.Task+<>c.<.cctor>b__277_0(System.Object) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(System.Threading.Thread, System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef, System.Threading.Thread) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.Tasks.Task.ExecuteEntryUnsafe(System.Threading.Thread) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.Tasks.Task.ExecuteFromThreadPool(System.Threading.Thread) [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading.ThreadPoolWorkQueue.Dispatch() [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback() [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
System.NullReferenceException: Object reference not set to an instance of an object. [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PEMethodSymbol.get_ExplicitInterfaceImplementations() in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Metadata\PE\PEMethodSymbol.cs:line 1232 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SymbolExtensions.GetExplicitInterfaceImplementations(Symbol member) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\MemberSymbolExtensions.cs:line 516 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.MakeExplicitInterfaceImplementationMap() in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\TypeSymbol.cs:line 2091 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.GetExplicitImplementationForInterfaceMember(Symbol interfaceMember) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\TypeSymbol.cs:line 2082 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.ComputeImplementationForInterfaceMember(Symbol interfaceMember, TypeSymbol implementingType, DiagnosticBag diagnostics, Boolean ignoreImplementationInInterfaces, Boolean& implementationInInterfacesMightChangeResult) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\TypeSymbol.cs:line 839 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.ComputeImplementationAndDiagnosticsForInterfaceMember(Symbol interfaceMember, Boolean ignoreImplementationInInterfaces, Boolean& implementationInInterfacesMightChangeResult) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\TypeSymbol.cs:line 781 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.FindImplementationForInterfaceMemberInNonInterfaceWithDiagnostics(Symbol interfaceMember, Boolean ignoreImplementationInInterfacesIfResultIsNotReady) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\TypeSymbol.cs:line 757 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.ComputeInterfaceImplementations(DiagnosticBag diagnostics, CancellationToken cancellationToken) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Source\SourceMemberContainerSymbol_ImplementationChecks.cs:line 157 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.GetSynthesizedExplicitImplementations(CancellationToken cancellationToken) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Source\SourceMemberContainerSymbol_ImplementationChecks.cs:line 50 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceMemberContainerTypeSymbol.ForceComplete(SourceLocation locationOpt, CancellationToken cancellationToken) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Source\SourceMemberContainerSymbol.cs:line 507 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbol.ForceCompleteMemberByLocation(SourceLocation locationOpt, Symbol member, CancellationToken cancellationToken) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Symbol.cs:line 800 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
   at Microsoft.CodeAnalysis.CSharp.Symbols.SourceNamespaceSymbol.<>c__DisplayClass49_1.<ForceComplete>b__0(Int32 i) in C:\Users\rikki\src\roslyn\src\Compilers\CSharp\Portable\Symbols\Source\SourceNamespaceSymbol_Completion.cs:line 61 [C:\Users\rikki\src\runtime\src\libraries\System.Diagnostics.Process\ref\System.Diagnostics.Process.csproj]
```

**Comment by RikkiGibson** [comment]
FYI @safern

**Comment by RikkiGibson** [comment]
OK, for a bit there we weren't sure if this bug was still present. But I think this CI run proves that it is: https://dev.azure.com/dnceng/public/_build/results?buildId=772143&view=logs&jobId=a2760323-e082-593b-18aa-f95a2816e578&j=a2760323-e082-593b-18aa-f95a2816e578&t=0fbc4e79-59cd-554b-7461-d6af1a16e28d

**Comment by RikkiGibson** [comment]
Things I have figured out about this bug:

- This bug was introduced between official builds 3.8.0-2.20403.10 (b0bff92d46d5cb8a495b3a6babb4ad59a8bdba50) and 3.8.0-2.20404.11 (2c5f58620006a58e72ac66ae266ae89c2744476c). [Compare link](https://github.com/dotnet/roslyn/compare/b0bff92d46d5cb8a495b3a6babb4ad59a8bdba50...2c5f58620006a58e72ac66ae266ae89c2744476c)
- This bug only seems to occur in Release mode, with no debugger attached to VBCSCompiler
- We only seem to crash when building ref assemblies such as [DbCommand in System.Data.Common](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Data.Common/ref/System.Data.Common.cs#L1845)

The compare link above is quite large and tends to time out, so I also have a list of PRs merged between the two builds


- [Merge pull request #46460 from dotnet/merges/master-to-master-vs-deps](https://www.github.com/dotnet/roslyn/pull/46460)
- [Merge pull request #46540 from dotnet/dev/jorobich/revert-formPerf](https://www.github.com/dotnet/roslyn/pull/46540)
- [Enable `with` expression on generic type paramteres constrained to a record type. (#46531)](https://www.github.com/dotnet/roslyn/pull/46531)
- [Disallow opening documents in remote workspace (#46523)](https://www.github.com/dotnet/roslyn/pull/46523)
- [Merge pull request #46521 from sharwell/squiggles](https://www.github.com/dotnet/roslyn/pull/46521)
- [Synthesize equality operators for records. (#46497)](https://www.github.com/dotnet/roslyn/pull/46497)
- [Disallow `static` modifier on records (#46517)](https://www.github.com/dotnet/roslyn/pull/46517)
- [Merge pull request #46504 from jmarolf/infra/do-not-run-CI-on-non-code-changes](https://www.github.com/dotnet/roslyn/pull/46504)
- [Add some tests from records test plan (#46466)](https://www.github.com/dotnet/roslyn/pull/46466)
- [Readonly fields have an accessible home when accessed from init-only setter (#46323)](https://www.github.com/dotnet/roslyn/pull/46323)
- [Merge pull request #45806 from dotnet/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45806)
- [Merge pull request #46484 from sharwell/document-token](https://www.github.com/dotnet/roslyn/pull/46484)
- [Merge pull request #46479 from akhera99/parameter-hints-experience-changes](https://www.github.com/dotnet/roslyn/pull/46479)
- [Produce warnings for definite assignment issues with private fields from metadata. (#46221)](https://www.github.com/dotnet/roslyn/pull/46221)
- [Merge pull request #46489 from sharwell/simpler-condition](https://www.github.com/dotnet/roslyn/pull/46489)
- [Remove WorkspaceKind.Test (#46366)](https://www.github.com/dotnet/roslyn/pull/46366)
- [Don't offer Use Explicit Type for target typed new (#46469)](https://www.github.com/dotnet/roslyn/pull/46469)
- [Add nullable annotations to NullableWalker (#45640)](https://www.github.com/dotnet/roslyn/pull/45640)
- [Merge pull request #46421 from 333fred/update-swix](https://www.github.com/dotnet/roslyn/pull/46421)
- [Integrate the covariant returns C# 9.0 feature to master (#44447)](https://www.github.com/dotnet/roslyn/pull/44447)
- [Tweak diagnostics to account for records (#46341)](https://www.github.com/dotnet/roslyn/pull/46341)
- [Merge pull request #45850 from sharwell/safe-shutdown](https://www.github.com/dotnet/roslyn/pull/45850)
- [Diagnose precedence inversion in a warning wave (#46239)](https://www.github.com/dotnet/roslyn/pull/46239)
- [Remove PROTOTYPE tag (#45965)](https://www.github.com/dotnet/roslyn/pull/45965)
- [Merge pull request #46378 from jaredpar/refactor](https://www.github.com/dotnet/roslyn/pull/46378)
- [Only run a single pass of NullableWalker per-member (#46402)](https://www.github.com/dotnet/roslyn/pull/46402)
- [Merge pull request #46452 from jmarolf/bugfix/FrameListener](https://www.github.com/dotnet/roslyn/pull/46452)
- [Fix crash, and offer "declare as nullable" for tuple fields (#46437)](https://www.github.com/dotnet/roslyn/pull/46437)
- [Merge pull request #46446 from akhera99/advanced_search_page](https://www.github.com/dotnet/roslyn/pull/46446)
- [Cannot assign maybe-null value to TNotNull variable (#41445)](https://www.github.com/dotnet/roslyn/pull/41445)
- [Fix overload resolution to handle binary compat in the face of covariant returns (#46367)](https://www.github.com/dotnet/roslyn/pull/46367)
- [Remove feature flag used for testing "DoNotRequireRuntimeCovariantReturnsSupport" (#46390)](https://www.github.com/dotnet/roslyn/pull/46390)
- [Merge pull request #46285 from gafter/covariant-returns-e2e1](https://www.github.com/dotnet/roslyn/pull/46285)
- [methodimpl(.override) should be accompanied by [PreserveBaseOverridesAttribute] (#45870)](https://www.github.com/dotnet/roslyn/pull/45870)
- [Reduce cascaded WRN_MultipleRuntimeOverrideMatches in certain rare scenarios (#45849)](https://www.github.com/dotnet/roslyn/pull/45849)
- [Merge pull request #45887 from m-redding/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45887)
- [Merge pull request #45883 from m-redding/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45883)
- [Merge pull request #45882 from m-redding/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45882)
- [Merge pull request #45880 from m-redding/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45880)
- [Strengthen tests for covariant returns (#45824)](https://www.github.com/dotnet/roslyn/pull/45824)
- [Merge pull request #45843 from m-redding/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45843)
- [Merge pull request #45746 from m-redding/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45746)
- [Merge pull request #45698 from sigmachirality/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45698)
- [Covariant returns part 4 (#44025)](https://www.github.com/dotnet/roslyn/pull/44025)
- [Merge pull request #45595 from m-redding/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45595)
- [Merge pull request #45517 from jmarolf/enable-ci](https://www.github.com/dotnet/roslyn/pull/45517)
- [Merge pull request #45487 from m-redding/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45487)
- [Merge pull request #45433 from m-redding/feature/convertNameOf](https://www.github.com/dotnet/roslyn/pull/45433)
- [Merge pull request #43797 from gafter/covariant-returns-part3](https://www.github.com/dotnet/roslyn/pull/43797)
- [Merge pull request #43673 from gafter/covariant-returns-part2](https://www.github.com/dotnet/roslyn/pull/43673)
- [Merge pull request #43723 from gafter/covariant-returns-merge-2020-04-27](https://www.github.com/dotnet/roslyn/pull/43723)
- [Merge pull request #43576 from gafter/covariant-returns-part1](https://www.github.com/dotnet/roslyn/pull/43576)
- [Merge pull request #43654 from gafter/covariant-returns-merge-2020-04-24](https://www.github.com/dotnet/roslyn/pull/43654)
- [Merge pull request #43577 from gafter/covariant-returns-merge-2020-04-22](https://www.github.com/dotnet/roslyn/pull/43577)
- [Merge pull request #43159 from gafter/merge-covariant-returns-2020-04-07](https://www.github.com/dotnet/roslyn/pull/43159)

**Comment by RikkiGibson** [comment]
We have added a workaround to the compiler which we should remove once we and our customers are ready to consume a .NET 5 runtime that has the fix.

**Comment by safern** [comment]
@RikkiGibson and @jaredpar  just FYI aspnet/extensions is hitting this as well:

```
Exception occurred with following context:

Compilation: Microsoft.AspNetCore.Testing.Tests

ISymbol: TestEventSource (NamedType)

System.NullReferenceException: Object reference not set to an instance of an object.

at Microsoft.CodeAnalysis.CSharp.Symbols.Metadata.PE.PEMethodSymbol.get_ExplicitInterfaceImplementations()

at Microsoft.CodeAnalysis.CSharp.Symbols.SymbolExtensions.GetExplicitInterfaceImplementations(Symbol member)

at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.MakeExplicitInterfaceImplementationMap()

at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.GetExplicitImplementationForInterfaceMember(Symbol interfaceMember)

at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.ComputeImplementationForInterfaceMember(Symbol interfaceMember, TypeSymbol implementingType, DiagnosticBag diagnostics, Boolean ignoreImplementationInInterfaces, Boolean& implementationInInterfacesMightChangeResult)

at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.ComputeImplementationAndDiagnosticsForInterfaceMember(Symbol interfaceMember, Boolean ignoreImplementationInInterfaces, Boolean& implementationInInterfacesMightChangeResult)

at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.FindImplementationForInterfaceMemberInNonInterfaceWithDiagnostics(Symbol interfaceMember, Boolean ignoreImplementationInInterfacesIfResultIsNotReady)

at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.FindImplementationForInterfaceMemberInNonInterface(Symbol interfaceMember, Boolean ignoreImplementationInInterfacesIfResultIsNotReady)

at Microsoft.CodeAnalysis.CSharp.Symbols.TypeSymbol.FindImplementationForInterfaceMember(Symbol interfaceMember)

at Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel.TypeSymbol.Microsoft.CodeAnalysis.ITypeSymbol.FindImplementationForInterfaceMember(ISymbol interfaceMember)
```

**Comment by RikkiGibson** [comment]
The latest compilers from the VS 16.8 channel should not have this issue. Is there a PR which updates the roslyn dependency for that repo?

**Comment by safern** [comment]
Yes, we're updating it to the same version as dotnet/runtime.

**Comment by RikkiGibson** [comment]
Could you please link the PR?

**Comment by safern** [comment]
This is one: https://github.com/dotnet/aspnetcore/pull/25545

@Anipik do you have the one for extensions?

**Comment by RikkiGibson** [comment]
That is surprising and unfortunate. We didn't think that the bug would reproduce for compilers that do not contain #44447. It might be necessary, unfortunately, to take a later compiler which includes #47007, the earliest of which is 3.8.0-3.20421.3

**Comment by safern** [comment]
FYI: @dougbu in case you don't want to downgrade

**Comment by dougbu** [comment]
(I hope to get dotnet/aspnetcore#25545 in as soon as I hear from my co-workers about the macOS timeouts we're also hitting and that I'm changing there. But, we can follow up later if necessary and I may not hear back today.)

Choosing a newer version of Microsoft.Net.Compilers.Toolset and leaving it unpinned in aspnetcore and aspnetcore-tooling is our current plan for the master branches now that arcade and aspnetcore use the same Mastro++ channels for this dependency (thanks @mmitche❕). But, based on our offline discussion about different versions used in the RC2 branches, I lean toward choosing a single version of this dependency for runtime, aspnetcore, aspnetcore-tooling, and extensions ASAP i.e. pinning them all at the same version (ignoring whatever arcade chooses from here on).

So, @Anipik @ericstj @safern will runtime and extensions move to something between `3.8.0-3.20421.3` (the "earliest" version @RikkiGibson mentioned) and `3.8.0-3.20452.1` (the latest in the 'VS 16.8' channel) today or tomorrow❔ If not, I'll leave dotnet/aspnetcore#25545 at `3.8.0-2.20403.2` for consistency w/ runtime's current toolset version.

**Comment by safern** [comment]
Since we don't need a compiler feature in runtime or extensions I don't think we will put up an update just to get a "newer" compiler in RC1/2 however we can do it in runtime or arcade master to dog food

**Comment by andrei-epure-sonarsource** [comment]
Hi. We opened in September #47187 which was closed as duplicate of this issue.

FYI #47187 doesn't reproduce anymore in dotnet `5.0.100-rc.2.20479.15` . I'm not sure if this issue should be closed or not.

**Comment by RikkiGibson** [comment]
I had meant to use this issue to track removing the workaround once we had a version of dotnet/runtime with the fix. We might be ready to do that now, so maybe I'll just give it a try and send the PR later this morning.
